jgrapht-core/src/test/java/org/jgrapht/alg/dual/LineGraphConstructorTest.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * (C) Copyright 2018-2018, by Nikhil Sharma and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.alg.dual;
+
+import org.jgrapht.*;
+import org.jgrapht.generate.GraphGenerator;
+import org.jgrapht.generate.StarGraphGenerator;
+import org.jgrapht.graph.*;
+import org.jgrapht.util.SupplierUtil;
+import org.junit.*;
+
+import java.util.*;
+
+import static org.junit.Assert.*;
+
+/**
+ * Tests for LineGraphConstructor
+ *
+ * @author Nikhil Sharma
+ */
+public class LineGraphConstructorTest
+{
+
+    @Test
+    public void testEmptyGraph()
+    {
+        // Line Graph of an empty graph should be empty
+        Graph<Integer, DefaultEdge> g = new SimpleGraph<>(DefaultEdge.class);
+        Graphs.addAllVertices(g, Arrays.asList(0, 1, 2, 3));
+
+        LineGraphConstructor<Integer, DefaultEdge, DefaultEdge> lgc = new LineGraphConstructor<>(g);
+        Graph<DefaultEdge, DefaultEdge> target = new SimpleWeightedGraph<>(DefaultEdge.class);
+        lgc.constructGraph(target);
+
+        assertTrue(GraphTests.isEmpty(g));
+    }
+
+    @Test
+    public void testStarGraph()
+    {
+        // Line Graph of a star graph is a complete graph
+        Graph<Integer, DefaultEdge> starGraph = new SimpleGraph<>(
+                SupplierUtil.createIntegerSupplier(), SupplierUtil.DEFAULT_EDGE_SUPPLIER, false);
+        GraphGenerator<Integer, DefaultEdge, Integer> generator = new StarGraphGenerator<>(5);
+        Map<String, Integer> resultMap = new HashMap<>();
+        generator.generateGraph(starGraph, resultMap);
+
+        LineGraphConstructor<Integer, DefaultEdge, DefaultEdge> lgc = new LineGraphConstructor<>(starGraph);
+        Graph<DefaultEdge, DefaultEdge> target = new SimpleGraph<>(DefaultEdge.class);
+        lgc.constructGraph(target);
+
+        assertTrue(GraphTests.isComplete(target));
+    }
+
+    @Test
+    public void testUndirectedGraph()
+    {
+        Graph<Integer, DefaultEdge> g = new SimpleGraph<>(DefaultEdge.class);
+        Graphs.addAllVertices(g, Arrays.asList(1, 2, 3, 4, 5));
+        g.addEdge(1,2);
```

Comment:
If you sue DefaultEdge e1=g.addEdge(1,2); then below you can use expectedGraph.addEdge(e1, e2);

Reply:
Done